.github/workflows/ci.yaml

```diff
@@ -0,0 +1,83 @@
+name: CI
+on: [push, pull_request]
+
+jobs:
+  clang-format:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: DoozyX/clang-format-lint-action@v0.12
+      with:
+        exclude: './thirdparty'
+        clangFormatVersion: 12
+        
+  build-ubuntu:
+    runs-on: ubuntu-latest
+    env:
+      dashboard_model: Experimental
+      build_type: Release
+      NUMBER_OF_PROCESSORS: 2
+      CXX: ${{ matrix.CXX }}
+    name: ${{ matrix.name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: build-ubuntu-gcc9
+            CXX: g++-9
+          - name: build-ubuntu-gcc10
+            CXX: g++-10
+          - name: build-ubuntu-clang10
+            CXX: clang++-10
+          - name: build-ubuntu-clang11
+            CXX: clang++-11
+            INSTALL_EXTRA: clang-11
+          - name: build-ubuntu-icpc
+            CXX: icpc
+            INSTALL_ONEAPI: true
+          - name: build-ubuntu-icpx
+            CXX: icpx
+            INSTALL_ONEAPI: true
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: install OneAPI
```

Comment:
Maybe we can just use their docker image? https://github.com/intel/oneapi-containers

Reply:
We could! I am not familiar with docker so I would need some help if we want to pursue this. However, what is wrong with just installing it via Intel's apt repository?

Reply:
For GitHub Actions, it should boil down to just adding `image: image=intel/oneapi-runtime` at the right place in the file...

Reply:
I just tried using their docker image, but then the submodule checkout fails: https://github.com/bernhardmgruber/Vc/runs/2885172514?check_suite_focus=true